Sources/Clappr/Classes/Base/Core.swift

```diff
@@ -120,16 +120,25 @@ open class Core: UIObject, UIGestureRecognizerDelegate {
 
     #if os(iOS)
     private func renderCoreAndMediaControlPlugins() {
-        plugins.forEach { plugin in
-            if isNotMediaControlPlugin(plugin) {
-                view.addSubview(plugin.view)
-                plugin.render()
-            }
+        renderCorePlugins()
+        renderMediaControlPlugins()
+    }
 
+    private func renderCorePlugins() {
+        plugins.filter { isNotMediaControlPlugin($0) }.forEach { plugin in
+            view.addSubview(plugin.view)
+            plugin.render()
+        }
+    }
 
-            if plugin is MediaControl, let mediaControl = plugin as? MediaControl {
-                mediaControl.renderPlugins(plugins)
-            }
+    private func renderMediaControlPlugins() {
+        let mediaControl = plugins.first { $0 is MediaControl }
```

Comment:
What is the behavior when we pass more than one Media Control? I mean, this architecture allows us to pass a Media Control in the plugins array and I'm not sure which one will be called.

Reply:
Well, at this moment it was assumed that only one MediaControl will be passed but if it changes and become necessary to handle multiple MediaControls it will be needed a more big and specific refactor to fix that.